File: Autobase/Backend/Autobase/src/main/java/com/example/Autobase/dto/response/driver/DriverResponseDto.java
package com.example.Autobase.dto.response.driver;

import lombok.Data;

import java.time.LocalDateTime;

@Data
public class DriverResponseDto {

    private Long id;
    private Long userId;
    private String name;
    private String phone;
    private String status;
    private int experienceYears;
    private String notes;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;

    public DriverResponseDto() {

    }

    public DriverResponseDto(Long id, Long userId, String name, String phone, String status, int experienceYears, String notes, LocalDateTime createdAt, LocalDateTime updatedAt) {
        this.id = id;
        this.userId = userId;
        this.name = name;
        this.phone = phone;
        this.status = status;
        this.experienceYears = experienceYears;
        this.notes = notes;
        this.createdAt = createdAt;
        this.updatedAt = updatedAt;
    }

}

File: Autobase/Frontend/Autobase/src/app/models/dto/response/driver/DriverResponseDto.ts
import { DriverStatus } from '../../../enums/DriverStatus';

export interface DriverResponseDto {
  id: number;
  userId: number;
  name: string;
  phone: string;
  status: DriverStatus;
  experienceYears: number;
  notes?: string;
  createdAt: string;
  updatedAt: string;
}
